monkeytype/typing.py

```diff
@@ -251,8 +251,19 @@ def rewrite(self, typ):
         return typ
 
 
+class RewriteGenerator(TypeRewriter):
+    """Returns an Iterator, if the send_type and return_type of a Generator is None"""
+
+    def visit_Generator(self, typ):
+        args = getattr(typ, '__args__')
+        for e in args:
+            if e.__args__[1] and e.__args__[2] is None:
```

Comment:
this doesn't do what you intend, because `is` binds more tightly than `and`; should be `e.__args__[1] is None and e.__args__[2] is None`